### PR TITLE
Disable controller on non console platforms

### PR DESCRIPTION
**Pull request by Northfear**
Just don't init it on anything else than Switch/Vita
Tested with Xbone One X controller on Linux

### CODE REVIEW

**Comment by ihhub** [comment]
Hi @Northfear , how about touchpads?

**Comment by Northfear** [comment]
Touchpad should be disabled now too (along with SW emulation when it's detected)

File: src/engine/localevent.cpp [hunk 1]
```diff
@@ -1153,8 +1155,6 @@ bool LocalEvent::HandleEvents( bool delay, bool allowExit )
         case SDL_MOUSEWHEEL:
             HandleMouseWheelEvent( event.wheel );
             break;
-
-#if defined( FHEROES2_VITA ) || defined( __SWITCH__ )
         case SDL_CONTROLLERDEVICEREMOVED:
```
Comment (ihhub) [inline src/engine/localevent.cpp:1158]: What will happen if a user adds a new controller?

**Review by ihhub — CHANGES_REQUESTED** [review]

File: src/engine/localevent.cpp [hunk 2]
```diff
@@ -1844,6 +1842,18 @@ void LocalEvent::SetStateDefaults( void )
     SetState( SDL_JOYHATMOTION, false );
     SetState( SDL_SYSWMEVENT, false );
 
+#if SDL_VERSION_ATLEAST( 2, 0, 0 )
```
[the remark this answers is not in the document]
Comment (Northfear) [reply src/engine/localevent.cpp:1845]: Touchpad events are SDL2 only. SDL 1.2 build just fails w/o this

File: src/engine/localevent.cpp (hunk 1 above)
Comment (Northfear) [reply src/engine/localevent.cpp:1158]: Nothing. Controller subsystem is not enabled and is not generation any controller-related events

File: src/engine/localevent.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (ihhub) [reply src/engine/localevent.cpp:1845]: My question was about the whole block. Why do we need this block?

[the remark this answers is not in the document]
Comment (Northfear) [reply src/engine/localevent.cpp:1845]: To disable generation of any touchpad-related events (there's only 3 of them: SDL_FINGERDOWN, SDL_FINGERUP and SDL_FINGERMOTION)

**Comment by ihhub** [comment]
@Northfear , thank you very much for fixing this issue!

**Pull request merged**